src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -25,118 +28,185 @@ export class ExamUpdateComponent implements OnInit {
+    }
+
+    get newWorkingTime(): number | undefined {
+        return this.exam.workingTime;
     }
 
     /**
      * Revert to the previous state, equivalent with pressing the back button on your browser
-     * Returns to the detail page if there is no previous state and we edited an existing exam
-     * Returns to the overview page if there is no previous state and we created a new exam
+     * Returns to the detail page if there is no previous state, and we edited an existing exam
+     * Returns to the overview page if there is no previous state, and we created a new exam
      */
-    previousState() {
+    resetToPreviousState() {
         this.navigationUtilService.navigateBackWithOptional(['course-management', this.course.id!.toString(), 'exams'], this.exam.id?.toString());
     }
 
+    /**
+     * Updates the working time for real exams based on the start and end dates.
+     */
+    updateExamWorkingTime() {
+        if (this.exam.testExam) return;
+
+        this.exam.workingTime = normalWorkingTime(this.exam);
+    }
+
+    /**
+     * Returns the maximum working time in minutes for test exams.
+     */
+    get maxWorkingTimeInMinutes(): number {
+        if (!this.exam.testExam) return 0;
+
+        if (this.exam.startDate && this.exam.endDate) {
+            return dayjs(this.exam.endDate).diff(this.exam.startDate, 'm');
+        } else {
+            // In case of an import, the exam.workingTime is imported, but the start / end date are deleted -> no error should be shown to the user in this case
+            return this.isImport ? this.workingTimeInMinutes : 0;
+        }
+    }
+
+    /**
+     * Saves the exam. If the dates have changed and the exam is ongoing, a confirmation modal is shown to the user.
+     * If either the user confirms the modal, the exam is not ongoing or the dates have not changed, the exam is saved.
+     */
+    handleSubmit() {
+        const datesChanged = this.exam.startDate?.diff(this.originalStartDate) !== 0 || this.exam.endDate?.diff(this.originalEndDate) !== 0;
+        if (datesChanged && this.isOngoingExam) {
+            const modalRef = this.modalService.open(ConfirmAutofocusModalComponent, { keyboard: true, size: 'lg' });
+            modalRef.componentInstance.title = 'artemisApp.examManagement.dateChange.title';
+            modalRef.componentInstance.text = this.artemisTranslatePipe.transform('artemisApp.examManagement.dateChange.message');
+            modalRef.componentInstance.contentRef = this.workingTimeConfirmationContent;
+            modalRef.result.then(this.save.bind(this));
+        } else {
+            this.save();
+        }
+    }
+
+    /**
+     * Saves the exam and navigates to the detail page of the exam if the save was successful.
+     * If the save was not successful, an error is shown to the user.
+     */
     save() {
         this.isSaving = true;
 
-        if (this.isImport) {
+        this.upsertOrImportExam()
```

Comment:
Please save this subscription reference and unsubscribe in ondestroy.

Reply:
Hm, the thing is, this is a short-lived subscription, not a websocket one or sth. It will be destroyed anyways after it either succeeds or fails. So the maximum duration a memory leak can exist is the network timeout, which is a few seconds. Also, I don't even know if rxjs has upstream cancellation built in, i.e. if the subscription is destroyed the network request is cancelled, which means, that the stuff will be kept in memory anyways as the network request is still in flight. So I can add it, but I guess it's not added in a lot of code pieces anyways.

Reply:
I added `takeWhile` now to both the http request subscription as well as the route subscription in `onInit`.